src/store/ChartStore.js

```diff
@@ -231,25 +231,67 @@ class ChartStore {
             }
 
             const onLayoutDataReady = () => {
-                this.restoreLayout(stxx, layoutData);
-
-                this.setActiveSymbols(active_symbols);
-
-                if (initialSymbol && !(layoutData && layoutData.symbols)) {
-                    this.changeSymbol(initialSymbol);
-                } else if (stxx.chart.symbol) {
-                    this.currentActiveSymbol = stxx.chart.symbolObject;
-                    stxx.chart.yAxis.decimalPlaces = stxx.chart.symbolObject.decimal_places;
-                    this.categorizedSymbols = this.categorizeActiveSymbols();
-                    if (onSymbolChange) { onSymbolChange(this.currentActiveSymbol); }
-                } else {
-                    this.changeSymbol(this.defaultSymbol);
-                }
-
+                const today_utc = Date.UTC(
+                        (new Date()).getUTCFullYear(),
+                        (new Date()).getUTCMonth(),
+                        (new Date()).getUTCDate(),
+                        (new Date()).getHours(),
+                        (new Date()).getMinutes(),
+                        (new Date()).getSeconds(),
+                    ),
+                    timeToEpochGMT = (time_stirng) => {
+                        let today = new Date(),
+                            time = time_stirng.split(':');
+                        return Date.UTC(today.getUTCFullYear(), today.getUTCMonth(), today.getUTCDate(), time[0], time[1], time[2]);
+                    },
+                    updateActiveSymbols = () => {
+                        api.getTradingTimes().then((data) => {
+                            data.trading_times.markets.forEach((market) => {
+                                market.submarkets.forEach((submarket) => {
+                                    submarket.symbols.forEach((symbol) => {
+                                        const foundSymbol = active_symbols.find(item => item.symbol === symbol.symbol);
```

Comment:
This is a pretty expensive operation. Consider that you are searching an array of 100 symbols 100 times every time you update the trading times. I think a better way would be to modify `exchange_is_open ` attribute directly from `categorizedSymbols` (also do it via an `action` for performance reasons).

It is more complex, but the overall structure of `data.trading_times` and `categorizedSymbols` are identical.